packages/run-protocol/test/vaultFactory/test-liquidator.js

```diff
@@ -0,0 +1,642 @@
+) {
+  const {
+    zoe,
+    runKit: { issuer: runIssuer, brand: runBrand },
+    aethKit: { brand: aethBrand, issuer: aethIssuer, mint: aethMint },
+    loanTiming,
+    rates,
+    aethInitialLiquidity,
+    runInitialLiquidity,
+  } = t.context;
+  t.context.timer = timer;
+
+  const runPayment = await getRunFromFaucet(t, runInitialLiquidity);
+  trace(t, 'faucet', { runInitialLiquidity, runPayment });
+  const runLiquidity = {
+    proposal: runInitialLiquidity,
+    payment: runPayment,
+  };
+  const aethLiquidity = {
+    proposal: aethInitialLiquidity,
+    payment: aethMint.mintPayment(aethInitialLiquidity),
+  };
+  const { amm: ammFacets, space } = await setupAmmAndElectorate(
+    t,
+    aethLiquidity,
+    runLiquidity,
+  );
+  const { consume, produce } = space;
+  trace(t, 'amm', { ammFacets });
+
+  const quoteMint = makeIssuerKit('quote', AssetKind.SET).mint;
+  // Cheesy hack for easy use of manual price authority
+  const priceAuthority = makeManualPriceAuthority({
+    actualBrandIn: aethBrand,
+    actualBrandOut: runBrand,
+    initialPrice: makeRatioFromAmounts(initialPrice, priceBase),
+    timer,
+    quoteMint,
+  });
+  produce.priceAuthority.resolve(priceAuthority);
+
+  const {
+    installation: { produce: iProduce },
+  } = space;
+  iProduce.VaultFactory.resolve(t.context.installation.VaultFactory);
+  iProduce.liquidate.resolve(t.context.installation.liquidate);
+  await startVaultFactory(space, { loanParams: loanTiming });
+
+  const governorCreatorFacet = consume.vaultFactoryGovernorCreator;
+  /** @type {Promise<VaultFactory & LimitedCreatorFacet<any>>} */
+  const vaultFactoryCreatorFacet = /** @type { any } */ (
+    E(governorCreatorFacet).getCreatorFacet()
+  );
+
+  // Add a vault that will lend on aeth collateral
+  const aethVaultManagerP = E(vaultFactoryCreatorFacet).addVaultType(
+    aethIssuer,
+    'AEth',
+    rates,
+  );
+
+  /** @type {[any, VaultFactory, VFC['publicFacet']]} */
+  // @ts-expect-error cast
+  const [governorInstance, vaultFactory, lender, aethVaultManager] =
+    await Promise.all([
+      E(consume.agoricNames).lookup('instance', 'VaultFactoryGovernor'),
+      vaultFactoryCreatorFacet,
+      E(governorCreatorFacet).getPublicFacet(),
+      aethVaultManagerP,
+    ]);
+  trace(t, 'pa', { governorInstance, vaultFactory, lender, priceAuthority });
+
+  const { g, v } = {
+    g: {
+      governorInstance,
+      governorPublicFacet: E(zoe).getPublicFacet(governorInstance),
+      governorCreatorFacet,
+    },
+    v: {
+      vaultFactory,
+      lender,
+      aethVaultManager,
+    },
+  };
+
+  return {
+    zoe,
+    // installs,
+    governor: g,
+    vaultFactory: v,
+    ammFacets,
+    runKit: { issuer: runIssuer, brand: runBrand },
+    priceAuthority,
+  };
+}
+// #endregion
+
+// sell in a single tranche
+//   - no shortfall
+//   - with shortfall
+// sell in multiple tranches
+// stall then oracle correct
+// stall then trade the amm back to the price
```

Comment:
notes to self?

Reply:
yep. tests to be added at the bottom. WIP.  Moving out